lib/datadog/core/telemetry/event.rb

```diff
@@ -1,81 +1,277 @@
 # frozen_string_literal: true
 
-require_relative 'collector'
-require_relative 'v1/app_event'
-require_relative 'v1/telemetry_request'
-require_relative 'v2/app_client_configuration_change'
-
 module Datadog
   module Core
     module Telemetry
-      # Class defining methods to construct a Telemetry event
       class Event
-        include Telemetry::Collector
+        # Base class for all Telemetry V2 events.
+        class Base
+          # The type of the event.
+          # It must be one of the stings defined in the Telemetry V2
+          # specification for event names.
+          def type; end
+
+          # The JSON payload for the event.
+          # @param seq_id [Integer] The sequence ID for the event.
+          def payload(seq_id)
+            {}
+          end
+        end
+
+        # Telemetry class for the 'app-started' event
+        class AppStarted < Base
+          def type
+            'app-started'
+          end
 
-        API_VERSION = 'v1'
```

Comment:
Is there any versioning of the API within our code?

Reply:
Yes, it's here: https://github.com/DataDog/dd-trace-rb/blob/fb2ca4dc49d813f0a61e27534d7d72dfd76e5088/lib/datadog/core/telemetry/http/ext.rb#L18